paddle/phi/core/compat/get_kerneltype_forvar_utils.h

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) 2023 PaddlePaddle Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include "paddle/phi/core/attribute.h"
+
+namespace phi {
+
+class KernelKey;
+class DenseTensor;
+/**
+ * Note: GetKernelTypeForVarContext is currently designed to MKLDNN kernel when
+ * the related memeber function 'GetKernelTypeForVar' is special. It is
+ * possiable to uesed for other custom hardwares in the future.
+ */
+class GetKernelTypeForVarContext {
+ public:
+  GetKernelTypeForVarContext() = default;
+  GetKernelTypeForVarContext(const GetKernelTypeForVarContext&) = default;
+  explicit GetKernelTypeForVarContext(const phi::KernelKey* kernel_key,
+                                      const AttributeMap* attrs)
+      : kernel_key_(kernel_key), attrs_(attrs) {}
+
+  const std::string& GetVarName(void) const;
+
+  const DenseTensor& GetTensor(void) const;
+
+  const KernelKey& GetKernelKey(void) const;
+
+  const AttributeMap& GetAttrs(void) const;
+
+  void SetVarName(std::string* var_name);
+
+  void SetDenseTensor(DenseTensor* tensor);
+
+ private:
+  const KernelKey* kernel_key_;  // not owned
+  // Use AttributeMap in namespace 'phi' to avoid depending 'fuild'
+  const AttributeMap* attrs_;  // not owned
```

Comment:
seems that oneDNN specific stuffs are hided inside the AttributeMap. Do you mind making these as members as this is only used for oneDNN kernel.

Reply:
Because the attribute name that will be used can be "data_layout" or "data_format" or other for the future vendors, it is brief to using AttributeMap.

Reply:
Personally, as data_layout/data_format has been defined through the [phi API](https://github.com/PaddlePaddle/Paddle/blob/develop/paddle/phi/common/layout.h#L36), it is natural to stay in the structure directly. I think you're right that `AttributeMap` will make it easier for the future vendors to store more specific stuffs, but at the same time, the content inside the map can be out of controlled by the framework. Both are okay to me.